overcloud.yml

```diff
@@ -231,9 +248,26 @@
             include_tasks: tasks/provisioning_baremetal_nodes.yml
         when: novaless_prov
 
+      - name: set fact for storage nodes
+        set_fact:
+          storage_node_disks: "{{ hostvars['undercloud']['storage_node_disks']|unique }}"
+        when: ceph_enabled and storage_node_disks is not defined
+
+      - name: generate internal.yml.j2
+        template:
+          src: internal.yml.j2
+          dest: ~/.infrared/plugins/tripleo-overcloud/templates/storage/internal.yml.j2
```

Comment:
instead of copying to infrared plugins directory, I recommend passing this file directly with --storage-config option. Please submit a new PR for that change.